magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,374 @@
+
+                        lstObsDir = (
+                            i[0].split("_")[0] + i[0].split("_")[1] + i[0].split("_")[2]
+                        )
+
+                        inputdir = f"/fefs/aswg/data/real/DL1/{lstObsDir}/{LST_version}/tailcut84"
+                        if not os.path.exists(
+                            f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}"
+                        ):
+                            os.mkdir(
+                                f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}"
+                            )
+                        if not os.path.exists(
+                            f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}/logs"
+                        ):
+                            os.mkdir(
+                                f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}/logs"
+                            )
+
+                        outputdir = f"{coincidence_DL1_dir}/DL1Coincident/{p}/NSB{nsb}/{lstObsDir}"
+                        list_of_subruns = np.sort(
+                            glob.glob(f"{inputdir}/dl1*Run*{i[1]}*.*.h5")
+                        )
+                        if os.path.exists(f"{outputdir}/logs/list_LST"):
+                            with open(
+                                f"{outputdir}/logs/list_LST", "a"
+                            ) as LSTdataPathFile:
+                                for subrun in list_of_subruns:
+                                    LSTdataPathFile.write(
+                                        f"{subrun}\n"
+                                    )  # If this files already exists, simply append the new information
+                        else:
+                            with open(
+                                f"{outputdir}/logs/list_LST.txt", "w"
+                            ) as f:  # If the file list_LST.txt does not exist, it will be created here
+                                for subrun in list_of_subruns:
+                                    f.write(f"{subrun}\n")
+
+                        if not os.path.exists(f"{outputdir}/logs/list_LST.txt"):
+                            continue
+                        process_size = (
+                            len(
+                                np.genfromtxt(
+                                    f"{outputdir}/logs/list_LST.txt", dtype="str"
+                                )
+                            )
+                            - 1
+                        )
+
+                        if process_size < 0:
+                            continue
+                        lines = [
+                            "#!/bin/sh\n\n",
+                            "#SBATCH -p short\n",
+                            f'#SBATCH -J {target_dir.split("/")[-2:][1]}_coincidence_{nsb}\n',
+                            f"#SBATCH --array=0-{process_size}\n",
+                            "#SBATCH --mem=30g\n",
```

Comment:
30 GB is a lot. I normally run it with 4 GB, however for some runs it is not enough. We could keep e.g. 8 GB as a default. A more clever system would run it first with size of the memory that is fine for most of the runs and if it fails the jot is resubmitted with twice larger memory